yt_dlp/extractor/reddit.py

```diff
@@ -148,6 +152,30 @@ def add_thumbnail(src):
             'age_limit': age_limit,
         }
 
+        # Check for embeds in text posts, or else raise to avoid recursing into the same reddit URL
+        if 'reddit.com/' in video_url and f'/{video_id}/' in video_url:
```

Comment:
use urllib.parse for better matching. reddit.com shouldn't match in query, for example